### PR TITLE
tests: tests no longer create instances of Basic with non-Basic args

### LINKED ISSUE #22549: Tests using Basic with non-Basic arguments
Currently, several tests are using code like `Basic(1)` or `Basic([])`. This creates `Basic` objects with non-`Basic` arguments, which should not be allowed. I suggest to change these to use their respective `Atom` object (e.g. `Int`, `Str` and `Tuple`), mostly because it sets a bad example/precedence.

<details>
<summary>Here is a list I found so far</summary>

```
  File "sympy/core/tests/test_basic.py", line 30, in test__aresame
  File "sympy/core/tests/test_basic.py", line 258, in test_atomic

  File "sympy/core/tests/test_exprtools.py", line 212, in test_gcd_terms

  File "sympy/core/tests/test_subs.py", line 629, in test_issue_6079

  File "sympy/printing/tests/test_dot.py", line 13, in test_purestr
  File "sympy/printing/tests/test_dot.py", line 25, in test_styleof

  File "sympy/printing/tests/test_mathml.py", line 1257, in test_print_basic

  File "sympy/strategies/branch/tests/test_tools.py", line 24, in test_zero_ints
  File "sympy/strategies/branch/tests/test_tools.py", line 31, in test_split5

  File "sympy/strategies/branch/tests/test_traverse.py", line 10, in test_top_down_easy
  File "sympy/strategies/branch/tests/test_traverse.py", line 17, in test_top_down_big_tree
  File "sympy/strategies/branch/tests/test_traverse.py", line 29, in test_top_down_harder_function
  File "sympy/strategies/branch/tests/test_traverse.py", line 36, in test_sall

  File "sympy/strategies/tests/test_rl.py", line 8, in test_rm_id
  File "sympy/strategies/tests/test_rl.py", line 25, in test_flatten
  File "sympy/strategies/tests/test_rl.py", line 28, in test_unpack
  File "sympy/strategies/tests/test_rl.py", line 32, in test_sort

  File "sympy/strategies/tests/test_tools.py", line 14, in test_subs_empty

  File "sympy/strategies/tests/test_traverse.py", line 19, in test_sall
  File "sympy/strategies/tests/test_traverse.py", line 23, in test_bottom_up
  File "sympy/strategies/tests/test_traverse.py", line 36, in _test_global_traversal
  File "sympy/strategies/tests/test_traverse.py", line 28, in test_top_down
  File "sympy/strategies/tests/test_traverse.py", line 36, in _test_global_traversal
  File "sympy/strategies/tests/test_traverse.py", line 63, in test_top_down_once
  File "sympy/strategies/tests/test_traverse.py", line 69, in test_bottom_up_once

  File "sympy/unify/tests/test_rewrite.py", line 13, in test_simple
  File "sympy/unify/tests/test_rewrite.py", line 24, in test_simple_variables
  File "sympy/unify/tests/test_sympy.py", line 13, in test_deconstruct
  File "sympy/unify/tests/test_sympy.py", line 26, in test_construct
  File "sympy/unify/tests/test_sympy.py", line 30, in test_nested
  File "sympy/unify/tests/test_sympy.py", line 36, in test_unify
  File "sympy/unify/tests/test_sympy.py", line 44, in test_unify_variables
  File "sympy/unify/tests/test_sympy.py", line 47, in test_s_input
```
</details>

### PR DESCRIPTION
**Pull request by ThePauliPrinciple**
#### References to other Issues or PRs

Fixes #22549

#### Brief description of what is fixed or changed
There were several uses of e.g. `Basic(1)` in the tests. This should
not be allowed, since it creates a `Basic` with non-`Basic` arguments.
This PR fixes all such uses that are directly in the tests. However,
it appears that for some tests, there are places in SymPy where
such invalid instances of `Basic` are still created.

#### Release Notes

NO ENTRY

### CODE REVIEW

File: sympy/core/tests/test_basic.py [hunk 1]
```diff
@@ -255,7 +255,7 @@ def test_atomic():
     assert _atomic(g(x + h(x))) == {g(x + h(x))}
     assert _atomic(g(x + h(x)), recursive=True) == {h(x), x, g(x + h(x))}
     assert _atomic(1) == set()
-    assert _atomic(Basic(1,2)) == {Basic(1, 2)}
+    assert _atomic(Basic(S(1), S(2))) == set()
```
Comment (oscarbenjamin) [inline sympy/core/tests/test_basic.py:258]: Why has this changed?

**Comment by oscarbenjamin** [comment]
Looks good. Thanks!

File: sympy/core/tests/test_basic.py (hunk 1 above)
Comment (ThePauliPrinciple) [reply sympy/core/tests/test_basic.py:258]: It seems that `_atomic` takes into account the case of `Basic` with non-`Basic` arguments and considers them atoms.

Comment (ThePauliPrinciple) [reply sympy/core/tests/test_basic.py:258]: Actually, it's weird that it returns nothing, this seems more like a bug that is somehow exposed now

**Pull request merged**